onedal/data_management/csr_table/csr_table.cpp

```diff
@@ -0,0 +1,169 @@
+                             const dal::array<std::int64_t>& col_indices,
+                             const dal::array<std::int64_t>& row_offsets,
+                             std::int64_t col_count,
+                             dal::sparse_indexing sp) {
+        return csr_table::wrap<Type>(non_zeros, col_indices, row_offsets, col_count, sp);
+    }));
+}
+
+template <typename Table, typename... Types>
+inline void instantiate_csr_constructor(py::class_<Table>& py_table,
+                                        const std::tuple<Types...>* const = nullptr) {
+    static_assert(std::is_same_v<Table, csr_table>);
+    return detail::apply(
+        [&](auto type_tag) -> void {
+            using type_t = std::decay_t<decltype(type_tag)>;
+            instantiate_csr_constructor_impl<type_t>(py_table);
+        },
+        Types{}...);
+}
+
+detail::csr_table_iface* get_csr_table_iface_impl(detail::table_iface* table) {
+    return dynamic_cast<detail::csr_table_iface*>(table);
+}
+
+template <typename Table>
+inline std::shared_ptr<detail::csr_table_iface> get_interface(const Table& table) {
+    auto pimpl = detail::pimpl_accessor{}.get_pimpl(table);
+    auto csr_iface_ptr = get_csr_table_iface_impl(pimpl.get());
+    return std::shared_ptr<detail::csr_table_iface>{ pimpl, csr_iface_ptr };
+}
+
+template <typename Type>
+dal::array<Type> get_data(const dal::csr_table& table) {
+    const std::int64_t elem_size = //
+        detail::integral_cast<std::int64_t>(sizeof(Type));
+
+    auto impl = get_interface(table);
+    const auto count = table.get_non_zero_count();
+
+    dal::array<dal::byte_t> data = impl->get_data();
+
+    const std::int64_t size = //
+        detail::check_mul_overflow(count, elem_size);
+
+    if (size != data.get_count()) {
+        throw std::length_error("Incorrect data size");
+    }
+
+    if (data.has_mutable_data()) {
+        dal::byte_t* raw_ptr = data.get_mutable_data();
+        Type* ptr = reinterpret_cast<Type*>(raw_ptr);
+        return dal::array<Type>(data, ptr, count);
+    }
+    else {
+        const dal::byte_t* raw_ptr = data.get_data();
+        const Type* ptr = reinterpret_cast<const Type*>(raw_ptr);
+        return dal::array<Type>(data, ptr, count);
+    }
+}
+
+py::object get_data_array(const dal::csr_table& table) {
+    const table_metadata& meta = table.get_metadata();
+    const data_type dtype = meta.get_data_type(0l);
+
+    return detail::dispatch_by_data_type(dtype, [&](auto type_tag) -> py::object {
+        using type_t = std::decay_t<decltype(type_tag)>;
+        auto array = get_data<type_t>(table);
+        return py::cast(std::move(array));
+    });
+}
+
+void instantiate_csr_table(py::module& pm) {
+    instantiate_csr_indexing(pm);
+
+    constexpr const char name[] = "csr_table";
+    py::class_<dal::csr_table> py_csr_table(pm, name);
+
+    //py_csr_table.def(py::init<dal::table>());
+
+    // Need to be fixed. No such symbol in the library
```

Comment:
fixed on https://github.com/oneapi-src/oneDAL/pull/2571
TODO: Need to be checked